core/utils/backoff_ticker.go

```diff
@@ -0,0 +1,85 @@
+package utils
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/jpillora/backoff"
+)
+
+type BackoffTicker struct {
+	b         backoff.Backoff
+	timer     *time.Timer
+	C         chan time.Time
+	chStop    chan struct{}
+	isRunning bool
+	sync.RWMutex
+}
+
+func NewBackoffTicker(min, max time.Duration) BackoffTicker {
+	c := make(chan time.Time, 1)
+	return BackoffTicker{
+		b: backoff.Backoff{
+			Min: min,
+			Max: max,
+		},
+		C:      c,
+		chStop: make(chan struct{}),
+	}
+}
+
+// Starts the ticker
+func (t *BackoffTicker) Start() {
+	t.Lock()
+	defer t.Unlock()
+
+	if t.isRunning {
+		return
+	}
+
+	// Reset the backoff
+	t.b.Reset()
+	go t.run()
+	t.isRunning = true
+}
+
+// Stop turns off the ticker
+func (t *BackoffTicker) Stop() {
```

Comment:
Noticing how you wrote this `Stop()` method -- do you want an instance of BackoffTicker to be restartable after being stopped? I might change/add feedback based on your intention here

Reply:
Yes it can be restarted by calling `Start` again on after Stop is called. This is similar to how `utils.PausableTicker` works